HW15/HW15.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <string>
+
+struct Student {
+    std::string name;
+    unsigned marks[4];
+};
+
+double averageMark(const Student& student) {
+    unsigned sum = 0;
+    for (int i = 0; i < 4; ++i) {
+        sum += student.marks[i];
+    }
+    return static_cast <double> (sum) / 4.0;
+}
+void bubbleSortStudents(Student students[], int numStudents) {
+    for (int i = 0; i < numStudents - 1; ++i) {
+        for (int j = 0; j < numStudents - i - 1; ++j) {
+            if (averageMark(students[j]) > averageMark(students[j + 1])) {
+                std::swap(students[j], students[j + 1]);
+            }
+        }
+    }
+}
+
+const Student* findMostSuccessfulStudent(const Student students[], int numStudents) {
+    const Student* mostSuccessful = &students[0];
+
+    for (int i = 1; i < numStudents; ++i) {
+        if (averageMark(students[i]) > averageMark(*mostSuccessful)) {
```

Comment:
no need to calculate average of mostSuccessful student every iteration.
can `cache` (read pre-save) mostSuccessfulMark and use it in comparison to reduce average calculations amount